vcd/datasource_not_found_test.go

```diff
@@ -250,8 +250,8 @@ func addMandatoryParams(dataSourceName string, mandatoryFields []string, t *test
 		if (dataSourceName == "vcd_nsxt_edgegateway_bgp_configuration" || dataSourceName == "vcd_nsxt_alb_settings" ||
 			dataSourceName == "vcd_nsxt_edgegateway_rate_limiting" || dataSourceName == "vcd_nsxt_edgegateway_dhcp_forwarding" ||
 			dataSourceName == "vcd_nsxt_firewall" || dataSourceName == "vcd_nsxt_route_advertisement" ||
-			dataSourceName == "vcd_nsxt_edgegateway_dhcpv6") &&
-			mandatoryFields[fieldIndex] == "edge_gateway_id" {
+			// #nosec G101 -- not a credential
+			dataSourceName == "vcd_nsxt_edgegateway_dhcpv6") || dataSourceName == "vcd_nsxt_edgegateway_dns" && mandatoryFields[fieldIndex] == "edge_gateway_id" {
```

Comment:
NIT and a matter of preference, but at least for me, readability is way better when this check is grouped with other data source names) as in the current state it makes me think how is `vcd_nsxt_edgegateway_dns` different from all the other data sources

```suggestion
			dataSourceName == "vcd_nsxt_edgegateway_dhcpv6" || dataSourceName == "vcd_nsxt_edgegateway_dns") && mandatoryFields[fieldIndex] == "edge_gateway_id" {
```

Reply:
Oops, missed the parentheses, thanks for spotting, fixed.